weight-meter/weight-meter-procedural/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+#![allow(unused_imports)]
+
+use proc_macro::TokenStream;
+use quote::{format_ident, quote, ToTokens};
+use syn::{Attribute, FnArg, Ident, ImplItem, ImplItemMethod, Item, ItemFn, ItemImpl, ItemMod, Pat};
+
+#[proc_macro_attribute]
+pub fn start_with(attr: TokenStream, item: TokenStream) -> TokenStream {
+	let base_weight: syn::Expr = syn::parse(attr).unwrap();
+	let ItemFn { attrs, vis, sig, block } = syn::parse(item).unwrap();
+	(quote! {
+		#(#attrs)*
+		#vis #sig {
+			orml_weight_meter::start_with(#base_weight);
```

Comment:
should avoid hardcode crate name in macro. should be something like `$crate::orml_weight_meter::start_with`

Reply:
yeah makes sense

Reply:
Seems like this works only for macros. Procedural macros are unhygienic https://doc.rust-lang.org/reference/procedural-macros.html

Reply:
This is how substrate solves it https://github.com/paritytech/substrate/blob/master/frame/support/procedural/tools/src/lib.rs